osbrain/tests/test_agent.py

```diff
@@ -89,22 +88,22 @@ def run(self):
     assert a0.ping() == 'pong'
 
 
-def test_agent_shutdown(nsaddr):
+def test_agent_shutdown(nsproxy):
     """
     An agent must unregister itself before shutting down.
     """
-    agent = AgentProcess('a0', nsaddr)
+    agent = AgentProcess('a0', nsproxy.addr())
     agent.start()
-    a0 = Proxy('a0', nsaddr)
+    a0 = Proxy('a0', nsproxy.addr())
     a0.run()
-    ns = NSProxy(nsaddr)
+    ns = NSProxy(nsproxy.addr())
```

Comment:
You don't really need to declare `ns`, as you now already have `nsproxy`. :wink: